apps/web/src/pages/integrations/components/multi-provider/UpdateProviderSidebar.tsx

```diff
@@ -167,22 +167,28 @@ export function UpdateProviderSidebar({
       return;
     }
 
-    const { channel: selectedChannel, environmentId, primary } = selectedProvider;
+    const { channel: selectedChannel, environmentId, primary, conditions } = selectedProvider;
     const isActiveFieldChanged = dirtyFields.active;
     const hasSameChannelActiveIntegration = !!providers
       .filter((el) => !NOVU_PROVIDERS.includes(el.providerId) && el.integrationId !== selectedProvider.integrationId)
       .find((el) => el.active && el.channel === selectedChannel && el.environmentId === environmentId);
     const isChannelSupportPrimary = CHANNELS_WITH_PRIMARY.includes(selectedChannel);
 
-    if (
-      isActiveFieldChanged &&
-      isChannelSupportPrimary &&
-      ((isActive && hasSameChannelActiveIntegration) || (!isActive && primary && hasSameChannelActiveIntegration))
-    ) {
+    const isChangedToActive =
+      isActiveFieldChanged && isChannelSupportPrimary && isActive && hasSameChannelActiveIntegration;
+
+    const isChangedToInactiveAndIsPrimary =
+      isActiveFieldChanged && isChannelSupportPrimary && !isActive && primary && hasSameChannelActiveIntegration;
+
+    const isPrimaryAndHasConditionsApplied = primary && conditions && conditions.length > 0;
```

Comment:
```suggestion
    const isPrimaryAndHasConditionsApplied = primary && conditions && conditions.length > 0 && hasSameChannelActiveIntegration;
```
forgot to mention that there is no sense to show empty modal if no integrations ;)